src/Grid/topology.jl

```diff
@@ -86,12 +110,31 @@ function _get_facet_facet_neighborhood(::ExclusiveTopology, #=rdim=#::Val{:mixed
     Access the `vertex_vertex_neighbor`, `edge_edge_neighbor`, or `face_face_neighbor` fields explicitly instead."))
 end
 
-function Base.show(io::IO, ::MIME"text/plain", topology::ExclusiveTopology)
-    println(io, "ExclusiveTopology\n")
-    print(io, "  Vertex neighbors: $(size(topology.vertex_vertex_neighbor))\n")
-    print(io, "  Face neighbors: $(size(topology.face_face_neighbor))\n")
-    println(io, "  Edge neighbors: $(size(topology.edge_edge_neighbor))")
+# Guess of how many neighbors depending on grid dimension and index type.
+# Better to guess a bit too high than a bit too low.
+function _getsizehint(g::AbstractGrid, ::Type{IDX}) where IDX
+    CT = getcelltype(g)
+    isconcretetype(CT) && return _getsizehint(getrefshape(CT)(), IDX)
+    rdim = get_reference_dimension(g)::Int
+    return _getsizehint(RefSimplex{rdim}(), IDX) # Simplex is "worst case", used as default.
 end
+_getsizehint(::AbstractRefShape,    ::Type{FaceIndex}) = 1 # Always 1 or zero if not mixed rdim
+
+_getsizehint(::AbstractRefShape{1}, ::Type{EdgeIndex}) = 1
+_getsizehint(::AbstractRefShape{2}, ::Type{EdgeIndex}) = 1
+_getsizehint(::AbstractRefShape{3}, ::Type{EdgeIndex}) = 3 # Number for RefTetrahedron
+_getsizehint(::RefHexahedron,       ::Type{EdgeIndex}) = 1 # Optim for RefHexahedron
```

Comment:
Can you add some commen on why this is optimal for hex? I am not seeing it directly, sorry.

Reply:
I don't think it is easy to see due to the exclusive nature, I could add a ref to https://github.com/Ferrite-FEM/Ferrite.jl/pull/974#issuecomment-2182897133 if you think that helps?

Reply:
Ohh, you mean for the grids generate with `generate_grid` optimal? 

Have you by any chance tried grids coming from external tools like gmsh then?

Reply:
No, I found that to be out of scope for this PR since this is just an optimization. Of course for non-structured grids, there will be cases when the numbers are higher for some entities, but we probably don't won't these sizehints for the worst case as that will over-allocate. From what I've seen during development, the key is to have most connections below the sizehint. 

From my perspective, if someone wants to do a proper investigation into this, that can be done at any point, since changing these sizehints only changes the performance.